type-check/throwIf.test.ts

```diff
@@ -0,0 +1,27 @@
+import { isNil, pipe, throwIf } from "../src";
+import * as Test from "../src/types/Test";
+
+const { checks, check } = Test;
+
+interface User {
+  name?: string | null;
+  age: number;
+}
+
+const createUser = (): User => ({ name: null, age: 23 });
+
+const res1 = createUser();
+const res2 = pipe(
+  createUser(),
+
+  (user) => user.name,
+
+  throwIf<string | null | undefined, null | undefined>(isNil, (name) =>
+    Error(`User name is ${typeof name}`),
+  ),
+);
+
```

Comment:
Suggestion: It seems better to modify the function to be immediately executable when all the required arguments are passed, as it appears in the examples below (`res3` and `res4`).
Additionally, since it supports variadic arguments, it seems necessary to add test cases for that as well.

```ts
// number[]
const res3 = throwIf(
  (list) => list.includes(0),
  () => new Error("0"),
  [1, 2, 3],
);

// number[]
const res4 = throwIf((list) => list.includes(0), [1, 2, 3]);

// number[]
const res5 = pipe(
  [1, 2, 3],
  throwIf((list) => list.includes(0)),
);

// number[]
const res6 = pipe(
  [1, 2, 3],
  throwIf(
    (list) => list.includes(0),
    () => new Error("0"),
  ),
);

// 3 | 1
const res7 = pipe(
  ([1, 2, 3] as const)[Math.random()],
  throwIf((a): a is 2 => a === 2),
);

// 3 | 1
const res8 = pipe(
  ([1, 2, 3] as const)[Math.random()],
  throwIf(
    (a): a is 2 => a === 2,
    () => new Error("0"),
  ),
);
```